services/m3dbnode/main/main.go

```diff
@@ -23,114 +23,383 @@ package main
 import (
 	"flag"
 	"fmt"
+	"io"
+	"log"
+	"math"
+	"net/http"
 	_ "net/http/pprof"
 	"os"
 	"os/signal"
+	"runtime"
+	"runtime/debug"
 	"syscall"
 	"time"
 
+	etcdclient "github.com/m3db/m3cluster/client/etcd"
+	"github.com/m3db/m3cluster/generated/proto/commonpb"
+	"github.com/m3db/m3cluster/kv"
+	"github.com/m3db/m3cluster/kv/util"
+	"github.com/m3db/m3cluster/services"
 	"github.com/m3db/m3db/client"
-	"github.com/m3db/m3db/services/m3dbnode/server"
+	"github.com/m3db/m3db/context"
+	"github.com/m3db/m3db/encoding"
+	"github.com/m3db/m3db/encoding/m3tsz"
+	"github.com/m3db/m3db/kvconfig"
+	hjcluster "github.com/m3db/m3db/network/server/httpjson/cluster"
+	hjnode "github.com/m3db/m3db/network/server/httpjson/node"
+	"github.com/m3db/m3db/network/server/tchannelthrift"
+	ttnode "github.com/m3db/m3db/network/server/tchannelthrift/node"
+	"github.com/m3db/m3db/persist/fs"
+	"github.com/m3db/m3db/persist/fs/commitlog"
+	"github.com/m3db/m3db/ratelimit"
+	"github.com/m3db/m3db/retention"
+	m3dbruntime "github.com/m3db/m3db/runtime"
+	"github.com/m3db/m3db/services/m3dbnode/config"
 	"github.com/m3db/m3db/storage"
+	"github.com/m3db/m3db/storage/block"
 	"github.com/m3db/m3db/storage/cluster"
 	"github.com/m3db/m3db/storage/namespace"
+	"github.com/m3db/m3db/storage/repair"
+	"github.com/m3db/m3db/storage/series"
+	"github.com/m3db/m3db/topology"
+	"github.com/m3db/m3db/ts"
+	m3dbxio "github.com/m3db/m3db/x/io"
+	"github.com/m3db/m3db/x/tchannel"
+	xconfig "github.com/m3db/m3x/config"
+	"github.com/m3db/m3x/instrument"
+	xlog "github.com/m3db/m3x/log"
+	"github.com/m3db/m3x/pool"
+
+	"github.com/uber-go/tally"
+)
+
+const (
+	bootstrapConfigInitTimeout = 10 * time.Second
+	serverGracefulCloseTimeout = 10 * time.Second
 )
 
 var (
-	idArg                  = flag.String("id", "", "Node host ID")
-	httpClusterAddrArg     = flag.String("clusterhttpaddr", "0.0.0.0:9000", "Cluster HTTP server address")
-	tchannelClusterAddrArg = flag.String("clustertchanneladdr", "0.0.0.0:9001", "Cluster TChannel server address")
-	httpNodeAddrArg        = flag.String("nodehttpaddr", "0.0.0.0:9002", "Node HTTP server address")
-	tchannelNodeAddrArg    = flag.String("nodetchanneladdr", "0.0.0.0:9003", "Node TChannel server address")
+	configFile = flag.String("f", "", "configuration file")
```

Comment:
could you include a sample config and write a test which verifies the service node is able to startup with the it

Reply:
I was thinking more about a test that verifies a YAML transforms to the given struct expected in the config package.

For integration test of an actual node coming up and being accessible we'll need to wait for an embedded etcd server solution, which will come in a subsequent change. i.e. ability to startup the node running an embedded etcd server.